---
title: "Differential expression"
output: 
  html_document:
    code_folding: hide
---

This Rmarkdown document presents results from a bulk RNA-seq experiment that has been processed using mapping (e.g. hisat2) and then features are counted using featurecounts. For data that has been processed using pseudoalignment please refer to DE_pseudoalignment.

To run the Rmarkdown please save your data as 


```{r, include=FALSE}
source('functions.R')
library(org.Hs.eg.db)
library(DESeq2)
library(pheatmap)
library(tidyverse)
library(DT)
library(limma)
library("ggplot2") #Best plots
library("ggrepel") #Avoid overlapping labels
```


```{r , echo=TRUE,warning=FALSE,message=FALSE,error=FALSE, include=FALSE}

design_files <- list.files(pattern = "design_")

if (file.exists("featurecounts.tsv.gz")) {
  df_mRNA <- read.table(gzfile("featurecounts.tsv.gz"), sep = "\t", header = TRUE, row.names = 1)
  colnames(df_mRNA) <- gsub(".", "-", x = colnames(df_mRNA), fixed = T)
} else {
  print("Please add featurecounts.tsv.gz into the project folder as it seems to be missing")
}


if (file.exists(design_files[1])) {
  for (i in design_files){
  meta_data <- read.table(i, sep=",", header = TRUE) 
  rownames(meta_data) <- meta_data$Sample
  df_mRNA = df_mRNA[,rownames(meta_data)]
  all(rownames(meta_data) %in% colnames(df_mRNA))
  assign(paste("meta_data", i, sep = "."), meta_data)
  }
} else {
    print("No design files were detected please add a file called design_<test>_<control>_<test>_<column>.csv. Please refer to documentation on github for more ifnormation")
  }

```

```{r , message=FALSE, include=FALSE}
## Set up the DESeq2 object

for (i in design_files) {
  x <- strsplit(i, "_")
  stat.test <- x[[1]][2]
  control <- x[[1]][3]
  test <- x[[1]][4]
  value <- x[[1]][5]
  value <- gsub(".csv","",value)
  meta_data <- get(gsub("SAMPLE_FILE",i , "meta_data.SAMPLE_FILE"))
  model <- as.character(meta_data$model[[1]])
  
  deseq_results <- run_deseq2(as.data.frame(df_mRNA), meta_data, control = control, test=test, value=value, model = model)
  
  res = deseq_results@res
  dds = deseq_results@dds
  assign(paste("res", i, sep = "."), res)
  assign(paste("dds", i, sep = "."), dds)
}

```

# Model fitting

This section of the report describes figures that can be used to assess how well the DESeq2 model has fitted the data.

## Dispersion {.tabset .tabset-fade}

Plotting the dispersion estimates is a useful diagnostic. The dispersion plot below is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value, (this outlier detection is described in the manual page for estimateDispersionsMAP). The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.


```{r, results='asis', echo = FALSE}
for (i in design_files){
  dds <- get(gsub("SAMPLE_FILE",i , "dds.SAMPLE_FILE"))
  cat("### ",i,"\n")
  plt <- plotDispEsts(dds)
  print(plt)
  cat('\n\n')
} 

```


## Summary of the data {.tabset .tabset-fade}


```{r, results='asis', echo = FALSE}
for (i in design_files){
  res <- get(gsub("SAMPLE_FILE",i , "res.SAMPLE_FILE"))
  cat("### ",i,"\n")
  summary(res, alpha=0.05)
  cat('\n\n')
} 

```


## MA plots {.tabset .tabset-fade}

In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.01. Points which fall out of the window are plotted as open triangles pointing either up or down.


```{r, results='asis', echo = FALSE}
for (i in design_files){
  res <- get(gsub("SAMPLE_FILE",i , "res.SAMPLE_FILE"))
  cat("### ",i,"\n")
  plt <- DESeq2::plotMA(res)
  print(plt)
  cat('\n\n')
} 

```

# Volcano plots {.tabset .tabset-fade}



```{r, results='asis', echo = FALSE}

for (i in design_files){
  res <- get(gsub("SAMPLE_FILE",i , "res.SAMPLE_FILE"))
  cat("### ",i,"\n")
  plt <- plot_volcano(res)
  print(plt)
  cat('\n\n')
} 


```

## Results tables

The folowing results tables show the significant genes. Filtering has been performed with a log2 fold change +/- 2.


### 24hr J4 vs DMSO

```{r , message=FALSE, echo=FALSE}
dt <- filter_genes(as.data.frame(res_24), name="featurecounts_24hrJ4vsdmso")

datatable(dt$sig)
```

### 6hr J4 vs DMSO

```{r , message=FALSE, echo=FALSE}
dt <- filter_genes(as.data.frame(res_6), name="featurecounts_6hrJ4vsdmso")

datatable(dt$sig)
```

### 3hr J4 vs DMSO

```{r , message=FALSE, echo=FALSE}
dt <- filter_genes(as.data.frame(res_3), name="featurecounts_3hrJ4vsdmso")

datatable(dt$sig)
```


# Conclusion

Given the high level of low count genes in the dataset: The biological and technical variation in the experiment dominates any true log fold changes across condition given your sample size. Another way to say this is that the experiment was underpowered to detect the changes across condition.